code/modules/mob/living/living_say.dm

```diff
@@ -332,10 +335,19 @@ GLOBAL_LIST_INIT(message_modes_stat_limits, list(
 	var/whisper_range = 0
 	var/is_speaker_whispering = FALSE
 	if(message_mods[WHISPER_MODE]) //If we're whispering
-		whisper_range = EAVESDROP_EXTRA_RANGE
+		whisper_range = 6 // Needed for good hearing trait. The actual filtering for whispers happens at the /mob/living/Hear proc
+		// 6 is the value needed to allow people with GOOD_HEARING to hear at the default message range of 7. This range doesn't
+		// actually have a define, nor does the range for whispers
 		is_speaker_whispering = TRUE
 
-	var/list/listening = get_hearers_in_view(message_range + whisper_range, source)
+	var/list/in_view = hearers(message_range + whisper_range, get_turf(source))
```

Comment:
this doesnt work because hearers only returns mobs, and some obj's can hear (for example radio that are set to transmit what they hear nearby to their radio channel). use get_hearers_in_view() here